### ISSUE TITLE
[BUG] Suport response.raw

### ISSUE THREAD

**Issue by 07pepa** [getsentry/responses#503]
presume folowing code 
```python
#this is corectly mocked wiht request and data is in body/text etc
r = requests.get('https://github.com/timeline.json', stream=True)
r.raw.decode_content=True
s3upload_fileobj(r.raw) ##a raw consumer from s3
 ```
 whatever consumer consumes is empty

This was noticed when trying to mock some logic that will download stuff from http to aws  and responses vere used in place of http server

moto was used in place of s3 but something simpler may be used. to test this

**Comment by beliaev-maksim** [comment]
please provide full self contained code that we can copy and run to reproduce the issue
please provide actual and expected result

**Comment by 07pepa** [comment]
so i tried to dumb it down and this fails for me

```python
import requests
import responses


@responses.activate
def test_raw():
    raw_body = "teststring"
    url = "http://test-raw.com"
    responses.add(responses.GET, url, body=raw_body)
    with requests.get(url, stream=True) as response:
        response.raise_for_status()
        raw = response.raw
        raw.decode_content = True

        assert raw.read() == raw_body.encode()
```

but it does not fail on second computer

so maybe it is caused by conda or python version that differs for me so on pc where it is not working it is 
python : 3.7.11 and 3.7.12
conda version: 4.11.0
requests 2.27.1

**Comment by beliaev-maksim** [comment]
it is working fine on latest master

![image](https://user-images.githubusercontent.com/51964909/154996829-f2d17169-8bab-4df9-9f67-25660108c491.png)

**Comment by 07pepa** [comment]
reinstalled anaconda.. solved
